src/api/app/models/event_subscription/find_for_event.rb

```diff
@@ -6,33 +6,34 @@ def initialize(event)
       @event = event
     end
 
-    def subscriptions
+    def subscriptions(channel = :instant_email)
       receivers_and_subscriptions = {}
 
       event.class.receiver_roles.flat_map do |receiver_role|
         # Find the users/groups who are receivers for this event
         receivers = event.send("#{receiver_role}s")
         receivers = filter_and_convert_groups_without_emails_to_users(receivers)
 
+        options = { eventtype: event.eventtype, receiver_role: receiver_role, channel: channel }
         # Find the default subscription for this eventtype and receiver_role
-        default_subscription = EventSubscription.defaults.find_by(eventtype: event.eventtype, receiver_role: receiver_role)
+        default_subscription = EventSubscription.defaults.find_by(options)
 
         receivers.each do |receiver|
           # Prevent multiple enabled subscriptions for the same subscriber & eventtype
           # Also skip if the receiver is the originator of this event
           next if receivers_and_subscriptions[receiver].present? || receiver == event.originator
 
           # Try to find the subscription for this receiver
-          receiver_subscription = EventSubscription.for_subscriber(receiver).find_by(eventtype: event.eventtype, receiver_role: receiver_role)
+          receiver_subscription = EventSubscription.for_subscriber(receiver).find_by(options)
 
           if receiver_subscription.present?
             # Use the receiver's subscription if it exists
-            if receiver_subscription.enabled?
+            if receiver_subscription.enable?
```

Comment:
Why not `enabled` ? `enable` would be more a verb, or a method to set a state and `enabled` a state. So if i understood it right, you are asking for the `state` so `if subscription.enabled?` would make more sense, no?

Reply:
Good point, thanks :smile_cat:

Reply:
I applied the changes in the PR #9405